### ISSUE TITLE
Is it working for COCO train2017?

### ISSUE THREAD

**Issue by cooc-2002** [hristo-vrigazov/mmap.ninja#3]
Hi.
Thank for your cool work.

I have applied it to boost my training and I can successfully make val2017 RaggedMmap.
However, I get the following error to make train2017 RaggedMmap.

```
6143it [00:50, 120.75it/s]
Traceback (most recent call last):
  File "/gpfs-volume/PycharmProjects/APP_PyTorch/to_mmap.py", line 25, in <module>
    train_images = RaggedMmap.from_generator(
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/ragged.py", line 159, in from_generator
    return base.from_generator_base(out_dir=out_dir,
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/base.py", line 105, in from_generator_base
    memmap.extend(samples)
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/ragged.py", line 99, in extend
    numpy.extend(self.memmap, numpy_bytes_slices.buffer)
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/numpy.py", line 131, in extend
    extend_dir(Path(np_mmap.filename).parent, arr)
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/numpy.py", line 127, in extend_dir
    base.shape_to_file(new_shape, out_dir / f'shape.ninja')
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/base.py", line 60, in shape_to_file
    out_file.write(shape_to_bytes(shape))
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/base.py", line 46, in shape_to_bytes
    res.extend(int_to_bytes(axis_len, fmt=fmt))
  File "/home/user/.local/lib/python3.8/site-packages/mmap_ninja/base.py", line 12, in int_to_bytes
    return struct.pack(fmt, inp)
struct.error: 'L' format requires 0 <= number <= 4294967295
```

Is it working for large image sets such as MS-COCO train2017 or ImageNet training dataset?

**Comment by hristo-vrigazov** [comment]
It should work, this is a bug. Thank you for reporting it, I will investigate it.

**Comment by hristo-vrigazov** [comment]
Could you send me a link and a code sample so I can reproduce your problem/question easier?

**Comment by hristo-vrigazov** [comment]
Thanks for reporting this, the fix was easy.

Here's a notebook with the fixed version `mmap_ninja==0.2.3`:

https://colab.research.google.com/drive/158p8p4QrZ7vVxiVRC7v6cHBuixlnaNXM#scrollTo=RTMGNCl4ilsi

**Comment by hristo-vrigazov** [comment]
Closing this as it runs ok after the fix. @cooc-2002 Feel free to reopen if it happens with the new version :)
